pkgs/tools/inputmethods/ibus-engines/ibus-rime/default.nix

```diff
@@ -0,0 +1,44 @@
+{ stdenv, fetchFromGitHub, cmake, pkg-config, gdk-pixbuf, glib, ibus, libnotify
+, librime, brise }:
+
+stdenv.mkDerivation rec {
+  pname = "ibus-rime";
+  version = "1.4.0";
+
+  src = fetchFromGitHub {
+    owner = "rime";
+    repo = "ibus-rime";
+    rev = version;
+    sha256 = "0zbajz7i18vrqwdyclzywvsjg6qzaih64jhi3pkxp7mbw8jc5vhy";
+  };
+
+  buildInputs = [ gdk-pixbuf glib ibus libnotify librime brise ];
+  nativeBuildInputs = [ cmake pkg-config ];
+
+  makeFlags = [ "PREFIX=$(out)" ];
+  dontUseCmakeConfigure = true;
+
+  prePatch = ''
+    substituteInPlace Makefile \
+       --replace 'cmake' 'cmake -DRIME_DATA_DIR=${brise}/share/rime-data'
```

Comment:
You should be able to specify this in `cmakeFlags`, see https://github.com/NixOS/nixpkgs/pull/77877/files#diff-22b83f2a09917d6f17fa4fb028cd36fd499c7d046298d46202d60a2fea4dde9fR39-R43

in combination with https://github.com/NixOS/nixpkgs/pull/77877#discussion_r491836081

Reply:
The ibus-rime project used a weird mixture of Makefile and cmake.  I choose to use the make build system here.  Therefore I have to patch Makefile.  The other PR used cmake build system and that caused other dirt hacks.  I think using Makefile is cleaner  here.

Reply:
The issue with this is that it steps around the flags we pass to CMake like `cmakeBuildType`. I have opened https://github.com/rime/ibus-rime/pull/112 to allow us to use CMake for everything, including correct paths in the files.

Reply:
Great. So this will be resolved in next ibus-rime release.